app/auth/map-auth.js

```diff
@@ -1,15 +1,20 @@
-const isInRole = require('./is-in-role')
 const { administrator, processor, user, recommender, authoriser } = require('./permissions')
+const { superAdmins } = require('../config')
 
 const mapAuth = (request) => {
+  const { isAuthenticated, credentials } = request.auth
+  const { username } = credentials.account
+
   return {
-    isAuthenticated: request.auth.isAuthenticated,
-    isAnonymous: !request.auth.isAuthenticated,
-    isAdministrator: request.auth.isAuthenticated && isInRole(request.auth.credentials, administrator),
-    isProcessor: request.auth.isAuthenticated && isInRole(request.auth.credentials, processor),
-    isUser: request.auth.isAuthenticated && isInRole(request.auth.credentials, user),
-    isRecommender: request.auth.isAuthenticated && isInRole(request.auth.credentials, recommender),
-    isAuthoriser: request.auth.isAuthenticated && isInRole(request.auth.credentials, authoriser)
+    isAuthenticated,
+    isAdministrator: isAuthenticated && credentials.scope.includes(administrator),
+    isProcessor: isAuthenticated && credentials.scope.includes(processor),
+    isUser: isAuthenticated && credentials.scope.includes(user),
+    isRecommender: isAuthenticated && credentials.scope.includes(recommender),
+    isAuthoriser: isAuthenticated && credentials.scope.includes(authoriser),
+    isSuperAdmin: isAuthenticated &&
+      credentials.scope.includes(administrator) &&
```

Comment:
Has there been some update on this, that maybe I have missed? I thought the group of users allowed to do these functions was just an arbitrary list iven to us, that could possibly change in future - has someone specified they'll always also be in admin role?

Reply:
The original requirement was simply allow users with the role 'administrator', after that we layered on top the 'super admin' set. My understanding is that all the users listed already have the 'administrator' role but I'm getting Indy to check and confirm this.

My feeling was that we should also check this during the mapping as otherwise you could end up with a user who's admin role as been revoked but then still has super admin capabilities.